helix-core/src/main/java/org/apache/helix/HelixCloudProperty.java

```diff
@@ -69,16 +69,28 @@
   // Other customized properties that may be used.
   private Properties _customizedCloudProperties = new Properties();
 
+  public HelixCloudProperty() {
+    setCloudEnabled(false);
+  }
+
   /**
    * Initialize Helix Cloud Property based on the provider
    * @param
    */
   public HelixCloudProperty(CloudConfig cloudConfig) {
+    this();
+    populateFieldsWithCloudConfig(cloudConfig);
+  }
+
+  public void populateFieldsWithCloudConfig(CloudConfig cloudConfig) {
+    if (cloudConfig == null || !cloudConfig.isCloudEnabled()) {
+      reset();
```

Comment:
Can we create a default CloudConfig using `buildEmptyCloudConfig()` and populate the fields in cloudConfig to cloudProperty? 
It would be simpler I guess.

Reply:
It will still skip the assignments to the variables because `cloudConfig.isCloudEnabled` returns false. But this reminds me, maybe I should change the assignment logic in this method. Let me try and update the PR if it works.

Reply:
Updated! @xyuanlu